src/hotspot/share/utilities/macros.hpp

```diff
@@ -431,6 +431,17 @@
 #define NOT_WINDOWS(code) code
 #endif
 
+#if !defined(_WINDOWS)
+#ifndef POSIX
+#define POSIX
+#endif
+#define POSIX_ONLY(code) code
+#define NOT_POSIX(code)
```

Comment:
Not sure this makes sense. At the moment only Windows is not-POSIX. If in the future we had another non-POSIX platform then I find it very unlikely that it would use the same code as Windows. So I would not like to see NOT_POSIX used incorrectly where we really only mean WINDOWS.

Reply:
My intention was exactly that, using it to protect *Posix* code from compiling on non-Posix platforms. Using !Windows there is just as wrong and not as descriptive. Most !Windows cases in hotspot are not affected by this since they have nothing to do with Posix; but there are a few and there may be more. E.g. I am currently working on a patch to protect the program break on Unices. This means dealing with sbrk, and depending on which route I go a POSIX_ONLY macro would make perfect sense.

That POSIX is defined as !Windows is only incidental. If you prefer, I can list the posix operating systems here; I did not do that for brevity.